pkg/messaging/direct_messaging.go

```diff
@@ -333,6 +334,10 @@ func (d *directMessaging) getRemoteApp(appID string) (remoteApp, error) {
 		return remoteApp{}, err
 	}
 
+	if d.resolver == nil {
```

Comment:
Can you add a unit test for this?

Reply:
Done (there was no test at all for the method, so had to write one)

Reply:
Do we know when this d.resolver will be nil? And can this probleam be fixed or d.resolver will allways be nil after this problem happens? (it means that all service invoke will fail) If so, we need to add more clear log content to notify the user what happened here.

Reply:
See #5533 . It can be `nil` if it fails to be initialized - Dapr still inits, but we do show error messages at that time.

Reply:
OK. So if dapr succeed to init name resolver, this problem will be resolved and not be nil.